TM1py/Services/CellService.py

```diff
@@ -368,6 +370,57 @@ def write_dataframe(self, cube_name: str, data: 'pd.DataFrame', dimensions: Iter
                           use_changeset=use_changeset,
                           **kwargs)
 
+    @require_pandas
+    @manage_transaction_log
+    def write_dataframe_async(self, cube_name: str, data: 'pd.DataFrame',
+                              slice_size_of_dataframe: int, max_workers: int,
+                              increment: bool = False):
+        """ Write DataFrame into a cube using unbound TI processes in a multi-threading way. Requires admin permissions.
+        For a DataFrame with > 1,000,000 rows, this function will at least save half of runtime compared with `write_dataframe` function.
+        Column order must match dimensions in the target cube with an additional column for the values.
+        Column names are not relevant.
+        :param cube_name:
+        :param data: Pandas Data Frame
+        :param slice_size_of_dataframe: Number of rows for each DataFrame slice, e.g. 10000
```

Comment:
We could add a note here saying the `slice_size_of_dataframe` should be less than 16381 for the highest performance.

Reply:
Same as above.

Reply:
I think I disagree. I think it would be most efficient to break the data frame into multiple data frames of each 32k records or into n data frames with each a multiple of 32k records.

Like if a data frame of 320k records is passed and max_workers is 5, I think there are 2 sensible approaches:

A: break data frame into 10 dataframes of 32k records each -> the `write_dataframe` function is called 10 times. 10 unbound processes will be executed with each 32k statements.

B: break data frame into 5 dataframes of 64k records each -> the `write_dataframe` function is called 5 times. 10 unbound processes will be executed with each 32k statements.

Reply:
There should be another approach as well when `max_workers` is not provided:
A DataFrame with 320K records can be divided as follows:
1. 10 threads - 32K each
2. 20 threads - 16K each (This will be the fastest way for bigger orgs with tanks and bazooka)

Spawning threads will take extra time and I believe TM1py should handle these stuff by default as we can't expect everybody to be a tech geek. 
People with skills can still optimize it further by playing with the `max_workers`.

IMHO **threadpool** is very less effective compared to **processpool** as what we are doing here is a CPU expensive operation.

If we can also implement another method that can write using **ProcessPoolExecutor** that would be awesome.

Reply:
Interesting thoughts! It would be interesting to quantify how the `ProcessPoolExecutor` performs compared to `ThreadPoolExecutor`. 

Perhaps we can enhance the function (now or in a separate PR) to support both executors and let the user choose through an optional function argument?